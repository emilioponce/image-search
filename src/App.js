import React from 'react'
import ReactDOM from 'react-dom'

const App = () => {
  return <div>Image Search</div>
}

export default App
