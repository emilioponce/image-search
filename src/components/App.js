import React from 'react'
import ReactDOM from 'react-dom'

import Header from './Header'
import Main from './Main'
import Footer from './Footer'

const App = () => {
  return (
    <div>
      <Header />
      <Main />
      <Footer />
    </div>
  )
}

export default App
