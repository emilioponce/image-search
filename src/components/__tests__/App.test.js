import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import App from '../App'

// how to avoid JEST warnings -> https://github.com/facebook/jest/issues/5258
jest.mock('../Header', () => 'mock-header')
jest.mock('../Main', () => 'mock-main')
jest.mock('../Footer', () => 'mock-footer')

describe('App component', () => {
  const component = <App />
  it('Renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(component, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('Snapshot matchs', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
