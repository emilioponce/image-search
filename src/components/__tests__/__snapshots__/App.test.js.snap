// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Snapshot matchs 1`] = `
<div>
  <mock-header />
  <mock-main />
  <mock-footer />
</div>
`;
